### PR TITLE
Autoinstallation: prevent issues with duplicate IP address due to some networks (bsc#1226461)

### PR DESCRIPTION
**Pull request by meaksh**
## What does this PR change?

This PR fixes an issue seen when dealing with Autoinstallation and some of your systems contain `docker*`, `virbr*` or `cni-podman*` interfaces with the same IP address.

Then, when triggering autoinstallation for those systems, an ISE is raised (due duplicate IP address on Cobbler) and autoinstallation is not triggered properly:

```
...
Caused by: org.cobbler.XmlRpcException: XmlRpcException calling cobbler.
        at org.cobbler.CobblerConnection.invokeMethod(CobblerConnection.java:160) ~[rhn.jar:?]
        at org.cobbler.CobblerConnection.invokeTokenMethod(CobblerConnection.java:190) ~[rhn.jar:?]
        at org.cobbler.SystemRecord.invokeModify(SystemRecord.java:310) ~[rhn.jar:?]
        at org.cobbler.CobblerObject.modify(CobblerObject.java:339) ~[rhn.jar:?]
        at org.cobbler.CobblerObject.modify(CobblerObject.java:322) ~[rhn.jar:?]
        at org.cobbler.SystemRecord.setNetworkInterfaces(SystemRecord.java:822) ~[rhn.jar:?]
...
Caused by: redstone.xmlrpc.XmlRpcFault: <class 'ValueError'>:IP address duplicate found "172.17.0.1". Object with the conflict has the name "my-system-host:1"
        at redstone.xmlrpc.XmlRpcClient.handleResponse(XmlRpcClient.java:444) ~[redstone-xmlrpc-client.jar:?]
        at redstone.xmlrpc.XmlRpcClient.endCall(XmlRpcClient.java:376) ~[redstone-xmlrpc-client.jar:?]
        at redstone.xmlrpc.XmlRpcClient.invoke(XmlRpcClient.java:165) ~[redstone-xmlrpc-client.jar:?]
```

The issue is caused as Java is not properly excluding these interfaces when preparing the Cobbler systems for autoinstallation, where some types of interfaces are being excluded to avoid this.

As a workaround, if we set `allow_duplicate_ips: true` on `/etc/cobbler/settings.yaml` then this issue is not happening.

So, to summarize, this PR does the following fixes:
- Fix string to parse when Cobbler raises IP address exception, to avoid a general ISE as it was expected in these cases.
- Make `sumautil` Salt module able to identify "bridge" network devices, as this is needed for Java for proper matching of network device types.
- Do not process container network devices: `docker*` and `cni-podman*`, when preparing an autoinstallation.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/24580
Port(s): https://github.com/SUSE/spacewalk/pull/24815

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests" 
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: java/code/src/com/redhat/rhn/domain/server/NetworkInterface.java [hunk 1]
```diff
@@ -313,6 +313,14 @@ public boolean isBond() {
         return "bonding".equals(module);
     }
 
+    /**
+     * true if the nic is a container network
+     * @return true if the nic is a container network
+     */
+    public boolean isContainerNetwork() {
+        return ("bridge".equals(module) && (getName().startsWith("docker") || getName().startsWith("cni-podman")));
```
Comment (mbussolotto) [inline java/code/src/com/redhat/rhn/domain/server/NetworkInterface.java:321]: Is there any other way to do it?  I see 2 problems:
- `netavark` is missing (but probably you just need to add `getName().startsWith("cni-netavark")`
-  I don't think this would work for K3s (I know, it's not released yet, but we're trying to do our best to not introduce something that might work for podman and not for kubernetes)

Comment (meaksh) [reply java/code/src/com/redhat/rhn/domain/server/NetworkInterface.java:321]: Thanks for the review @mbussolotto !

Ok, good point about `netavark` and Kubernetes. I'll try to enhance the mechanism.

In the end, I just mimic here what we already do for the `virbr*` devices: https://github.com/uyuni-project/uyuni/pull/9057/files#diff-dd63c17a089292848e40dfb8c02de48edb6b619425601d9fe81d02aa2cf3ba5eR329

File: java/spacewalk-java.changes.meaksh.Manager-4.3-prevent-issues-on-duplicate-ips-bsc1226461 [hunk 2]
```diff
@@ -0,0 +1 @@
+- Autoinstallation: prevent issues with duplicate IP address due to some networks (bsc#1226461)
```
Comment (agraul) [inline java/spacewalk-java.changes.meaksh.Manager-4.3-prevent-issues-on-duplicate-ips-bsc1226461:1]: I would rephrase this slightly to make it clearer. "due to some networks" is hard to understand IMO.

```suggestion
- Autoinstallation: prevent "duplicate IP address" issues cause by container networks (bsc#1226461)
```

File: susemanager-utils/susemanager-sls/src/modules/sumautil.py [hunk 3]
```diff
@@ -117,9 +117,12 @@ def get_net_module(iface):
     """
     sysfspath = os.path.join(SYSFS_NET_PATH, iface, "device/driver")
 
-    return (
-        os.path.exists(sysfspath) and os.path.split(os.readlink(sysfspath))[-1] or None
-    )
+    if os.path.exists(sysfspath):
+        return os.path.split(os.readlink(sysfspath))[-1] or None
+    elif os.path.exists(os.path.join(SYSFS_NET_PATH, iface, "bridge")):
```
Comment (agraul) [inline susemanager-utils/susemanager-sls/src/modules/sumautil.py:122]: Can we give this a name? Above we use `sysfspath`, which is now ambiguous. For consistency it would be good to give both of them a name

**Review by agraul — COMMENTED** [review]
I agree that the approach taken on the Java side might require quite a bit of maintenance for keeping up with new strings to compare the "module" to.

Other than that, I'd like us to be a bit more consistent with the different branches in sumautil by giving case each an easy-to-understand name

File: susemanager-utils/susemanager-sls/src/modules/sumautil.py (hunk 3 above)
Comment (meaksh) [reply susemanager-utils/susemanager-sls/src/modules/sumautil.py]: fixed

**Review by mbussolotto — APPROVED** [review]
